apps/ws/src/socket/usecases/external-services-route/external-services-route.usecase.ts

```diff
@@ -64,18 +65,55 @@ export class ExternalServicesRoute {
       return;
     }
 
-    const unreadCount = await this.messageRepository.getCount(
-      command._environmentId,
-      command.userId,
-      ChannelTypeEnum.IN_APP,
-      { read: false },
-      { limit: 101 }
-    );
+    const isNotificationSeverityEnabled = await this.featureFlagsService.getFlag({
+      key: FeatureFlagsKeysEnum.IS_NOTIFICATION_SEVERITY_ENABLED,
+      defaultValue: false,
+      environment: { _id: command._environmentId },
+    });
+
+    const [unreadCount, severityCounts] = await Promise.all([
+      this.messageRepository.getCount(
+        command._environmentId,
+        command.userId,
+        ChannelTypeEnum.IN_APP,
+        { read: false },
+        { limit: 101 },
+        undefined,
+        'primary'
+      ),
+      isNotificationSeverityEnabled
+        ? await this.messageRepository.getCountBySeverity(
+            command._environmentId,
+            command.userId,
+            ChannelTypeEnum.IN_APP,
+            { read: false, snoozed: false },
+            { limit: 99 }
+          )
+        : [],
+    ]);
+
     const paginationIndication: IUnreadCountPaginationIndication =
       unreadCount > 100 ? { unreadCount: 100, hasMore: true } : { unreadCount, hasMore: false };
 
+    const counts = {
+      total: unreadCount,
+      severity: {
+        high: 0,
+        medium: 0,
+        low: 0,
+        none: 0,
+      },
+    };
+
+    for (const { severity, count } of severityCounts) {
+      if (severity in counts.severity) {
+        counts.severity[severity] = count;
+      }
+    }
+
     await this.wsGateway.sendMessage(command.userId, WebSocketEventEnum.UNREAD, {
       unreadCount: paginationIndication.unreadCount,
+      counts,
```

Comment:
the WS `unread_count_changed` event is not sending the new `counts` object with `total` and `severity` information; I didn't remove the old `unreadCount` as it would be a breaking change to old NC